compiler/rustc_infer/src/traits/util.rs

```diff
@@ -235,6 +236,37 @@ pub fn elaborate<'tcx, O: Elaboratable<'tcx>>(
     elaborator
 }
 
+pub fn elaborate_predicates_of<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    def_id: DefId,
+) -> Elaborator<'tcx, (ty::Predicate<'tcx>, Span)> {
+    elaborate(
+        tcx,
+        tcx.predicates_of(def_id).predicates.iter().map(|(p, sp)| (p.as_predicate(), *sp)),
+    )
+}
+
+pub fn filter_predicates<'tcx>(
```

Comment:
I think this function should get moved closer to the error reporting. It doesn't seem useful to me other than for making some `filter` calls in error reporting shorter.

Also, like, what's up with the name? `filter_predicates` plus the lack of documentation plus the fact that this is in the `util` module makes it incredibly unclear that this is meant to filter *outlives* regions only. It should be called `filter_outlives_regions` or something.

Also, the fact that it only filters `TypeOutlives` obligations and always takes `RegionOutlives` obligations seems somewhat unintuitive